docs/proposals/resource-metrics-api.md

```diff
@@ -0,0 +1,166 @@
+due to performance issues.
+
+#### Scheduler
+
+Scheduler in order to schedule best-effort pods requires node level resource usage metrics
+as an average aggreated across 1 minute (the window may change in the future).
+The metrics should be available for all resources supported in the scheduler.
+
+## Other considered use cases
+
+This paragraph describes the other considered use cases and explains why they are out
+of the scope MVP version.
+
+#### Custom metrics in HPA
+
+HPA requires the latest value of application level metrics.
+
+The effort is suspended and it's no clear whether application level metrics should be
+available in API server so the use case initially won't be supported.
+
+#### Ubernetes
+
+In order to schedule best-effort pods in Ubernetes cluster the Ubernetes scheduler
+requires the same metrics that the Kubernetes scheduler. To effort is in planning phase
+so it's out the scope of Resource Metrics API effort however the support should be sufficient
+once Kubernetes scheduler is supported.
+
+#### kubectl top
+
+This feature is not yet specified/implemented although it seems reasonable to support the following options:
+- ```kubectl top node <id>``` - list all pods running on the given node accompanied with theirs cpu/mem usage
+- ```kubectl top pod <id>``` - print cpu usage stats for the given pod
+- ```kubectl top pod --selector=<label-query>``` - list all pods defined by the label query accompanied with theirs cpu/mem usage
+
+Since there are not hard requirements in this case it will be not supported initally in the API although
+it will be probably possible to provide a reasonable implementation of the feature anyway.
+
+#### Kubernetes dashboard
+
+[Kubernetes dashboard](https://github.com/kubernetes/dashboard) in order to draw graps requires resource usage
+in timeseries format from relatively long period of time. The aggreations should be also possible on various levels
+including replication controllers, deployments, services, etc.
+
+Since the use case is complicated it will not be supported initally in the API and they will query Heapster
+directly using some custom API there.
+
+## Proposed API
+
+Initially the metrics API will be in a separate [API group](api-group.md) called ```metrics```.
+Later if we decided to have Node and Pod in different API groups also
+NodeMetrics and PodMetrics should be in different API groups.
+
+#### Schema
+
+The proposed schema is as follow. Each top-level object has `TypeMeta` and `ObjectMeta` fields
+to be compatible with Kubernetes API standards.
+
+```go
+type NodeMetrics struct {
+  unversioned.TypeMeta
+  ObjectMeta
+
+  Timestamp unversioned.Time
+  Window    unversioned.Duration
+  
+  Usage map[ResourceType]Resource.Quantity
+}
+
+type PodMetrics struct {
+  unversioned.TypeMeta
+  ObjectMeta
+
+  Timestamp unversioned.Time
+  Window    unversioned.Duration
+
+  Containers []ContainerMetrics
+}
```

Comment:
What do you mean by adding "95th as an aggregation"? I'm unsure why the users can't compute that. And if we do want to start offering aggregated stats, then we should serve a distribution (e.g., f(x)) rather than a specific value (e.g., f(0.95)).